src/cylinder.jl

```diff
@@ -33,8 +33,8 @@ function surfpt_nearby(x::SVector{N}, s::Cylinder{N}) where {N}
     pout = copysign(1.0,p) * s.a
     qout = lq≠0 ? q/lq : @SVector(zeros(N))  # qout is used only when lq ≠ 0 below
 
-    onbndp = abs(lp-s.h2) ≤ Base.rtoldefault(Float64) * s.h2
-    onbndq = abs(lq-s.r) ≤ Base.rtoldefault(Float64) * s.r
+    onbndp = abs(lp.-s.h2) .≤ Base.rtoldefault(Float64) .* s.h2
+    onbndq = abs(lq.-s.r) .≤ Base.rtoldefault(Float64) .* s.r
```

Comment:
I'm confused, isn't this already a scalar calculation?  What are the dots for?

Reply:
Sorry, I thought `r` and `h2` of `Cylinder` were vectors (because `r` is a vector in `Box`).  Reverting to the regular operators.